tests/org.eclipse.core.tests.resources/src/org/eclipse/core/tests/resources/ResourceTest.java

```diff
@@ -1000,11 +1000,7 @@ protected void tearDown() throws Exception {
 	 * Blocks the calling thread until autobuild completes.
 	 */
 	protected void waitForBuild() {
-		try {
-			Job.getJobManager().join(ResourcesPlugin.FAMILY_AUTO_BUILD, null);
-		} catch (OperationCanceledException | InterruptedException e) {
-			//ignore
-		}
+		((Workspace) getWorkspace()).getBuildManager().waitForBuild();
```

Comment:
The change is wrong. There are **two** jobs actually, not just one, and this is the reason we have waited for the job **family**. See `org.eclipse.core.internal.events.AutoBuildJob.noBuildJob`.

Reply:
i don't get your comment: all i did was moving code and adding autoBuildJob.wakeUp(0) - it still waits for the family

Reply:
Ignore that, I'm still confused by github review UI. I haven't seen the other change